packages/rag/src/tools/vector-query.ts

```diff
@@ -9,8 +11,42 @@ import type { RagTool } from '../utils';
 import { convertToSources } from '../utils/convert-sources';
 import type { VectorQueryToolOptions } from './types';
 
+/**
+ * Resolves option values with priority: runtime context > function option > static option > default
+ * Handles both synchronous and asynchronous option resolvers
+ */
+async function resolveOption<T>(
+  key: string,
+  runtimeContext: RuntimeContext,
+  option: T | ((params: { runtimeContext: RuntimeContext }) => Promise<T> | T) | undefined,
+  defaultValue?: T,
+  logger?: MastraLogger,
+): Promise<T | undefined> {
+  // Check runtime context first
+  const runtimeValue = runtimeContext.get(key);
+  if (runtimeValue !== undefined && runtimeValue !== null) {
+    if (logger) {
+      logger.warn(
+        `[VectorQueryTool] Using runtime context values is deprecated. Use dynamic arguments instead: https://mastra.ai/en/reference/tools/vector-query-tool#example-with-dynamic-arguments`,
+      );
+    }
+    return runtimeValue as T;
+  }
+
+  // Handle function options
+  if (typeof option === 'function') {
+    const fn = option as (params: { runtimeContext: RuntimeContext }) => Promise<T> | T;
+    const result = fn({ runtimeContext });
+    return result;
+  }
+
+  // Return static option value
+  return option ?? defaultValue;
+}
+
```

Comment:
Could we make this a helper function both graph-rag and vector-query tool use? It looks like the only difference is the warning.